File: Project RESTful/Project-RESTful/src/main/java/com/example/bikes/Project/RESTful/model/entity/Bike.java
package com.example.bikes.Project.RESTful.model.entity;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import javax.persistence.*;
import javax.validation.constraints.NotBlank;
import javax.validation.constraints.Size;
import java.io.Serializable;
import java.time.LocalDateTime;
import java.util.Date;

@NoArgsConstructor
@AllArgsConstructor
@Data
@Entity
@Table(name = "bike")
public class Bike implements Serializable {
    public enum Type { ROAD, MOUNTAIN, GRAVEL }

    @Id
    private String serial;

    @Enumerated(EnumType.STRING)
    @Column (unique = false, nullable = false)
    private Type type;

    @NotBlank(message = "no puede estar vacio ni contener espacios en blanco.")
    @Size(min = 1, max = 20,  message = "tiene que contener entre 1 y 20 carácteres.")
    @Column (unique = false, nullable = false)
    private String brand;

    @NotBlank(message = "no puede estar vacio ni contener espacios en blanco.")
    @Column (unique = false, nullable = false)
    private double weight;

    @NotBlank(message = "no puede estar vacio ni contener espacios en blanco.")
    @Column (unique = false, nullable = false)
    private double price;

    @Column(name = "purchasedate", unique = false, nullable = false)
    private LocalDateTime purchaseDate;

    @PrePersist
    public void purchaseDatePersist()
    {
        purchaseDate = LocalDateTime.now();
        System.out.println(purchaseDate.toString());
    }
}

File: bike-project-net-client/bike-project-net-client/bike-project-net-client/Bike.cs
using Newtonsoft.Json;
using Newtonsoft.Json.Converters;
using System;

namespace bike_project_net_client
{
    public class Bike
    {
        [JsonConverter(typeof(StringEnumConverter))]
        public enum types
        {
            ROAD,
            MOUNTAIN,
            GRAVEL,
        }

        private string serialField;
        private types typeField;
        private string brandField;
        private double priceField;
        private double weightField;
        private DateTime purchaseDateField;

        public Bike(string serialField, string brandField, double priceField, double weightField, DateTime purchaseDateField)
        {
            this.serialField = serialField;
            this.brandField = brandField;
            this.priceField = priceField;
            this.weightField = weightField;
            this.purchaseDateField = purchaseDateField;
        }

        public string brand
        {
            get
            {
                return this.brandField;
            }
            set
            {
                this.brandField = value;
            }
        }

       public double price
        {
            get
            {
                return this.priceField;
            }
            set
            {
                this.priceField = value;
            }
        }

        public DateTime purchaseDate
        {
            get
            {
                return this.purchaseDateField;
            }
            set
            {
                this.purchaseDateField = value;
            }
        }

        public string serial
        {
            get
            {
                return this.serialField;
            }
            set
            {
                this.serialField = value;
            }
        }

        public types type
        {
            get
            {
                return this.typeField;
            }
            set
            {
                this.typeField = value;
            }
        }

        public double weight
        {
            get
            {
                return this.weightField;
            }
            set
            {
                this.weightField = value;
            }
        }
    }
}
